### ISSUE TITLE
Wrong path to assets when built on windows

### ISSUE THREAD

**Issue by jonespen** [goldhand/sw-precache-webpack-plugin#11]
![image](https://cloud.githubusercontent.com/assets/230841/18274005/8a021864-7440-11e6-83c1-1101870a592d.png)

Any configuration I'm missing, or is there a bug? It's working fine when building on my mac, but our build server is running TeamCity/Windows.

My config: 

``` javascript
new SWPrecacheWebpackPlugin({
  cacheId: 'sw-precache-pwa',
  filename: 'sw-precache.js',
})
```

Isn't https://github.com/goldhand/sw-precache-webpack-plugin/blob/master/src/index.js#L66 this supposed to strip away this?

**Comment by goldhand** [comment]
Yup thats exactly what that should do. What is your output path in your webpack configuration? Does it log the same as `"C:/TeamCity/buildAgent/work/"` or something similar?

**Comment by jonespen** [comment]
I believe it logged something like `C://TeamCity//buildAgent//work//...`. (I'll confirm this tomorrow) Dunno if the double slashes is a windows thing

**Comment by terjebra** [comment]
I encountered the same problem and got around it by adding stripPrefix. Due to windows
I had replace the slashes with forwardslashes in the path to get it working.

My sw-precache-config:

const SW_PRECACHE_CONFIG = {
  cacheId: SERVICE_WORKER_CACHEID,
  filename: SERVICE_WORKER_FILENAME,
  stripPrefix: path.join(__dirname, 'dist').replace(/\\/g,"/")
};

**Comment by davidgilbertson** [comment]
Thanks @terjebra that worked for me.

**Comment by goldhand** [comment]
@terjebra @davidgilbertson I think this should be fixed in `v0.7.0` here: https://github.com/goldhand/sw-precache-webpack-plugin/blob/9646bcb7845f5672ba51dc7f0036849f88e08fe6/src/index.js#L87

Would either of you mind testing it?

**Comment by goldhand** [comment]
Just tested, does not work still. Need to add the `stripPrefix` for windows somehow

**Comment by nasushkov** [comment]
Hi, I have the same issue, but I guess the problem resides in sw-precache and how it processes stripPrefix

**Comment by matthiasg** [comment]
@terjebra's example is expecting the configuration to be located in the same folder as the dist folder. make sure path.join(__dirname, 'dist') actually does point to your dist folder (or whatever you have as output folder)

**Comment by matthiasg** [comment]
sw-precache-webpack-plugin also automatically sets `stripPrefix` to webpack's `outputPath` and is influenced by the `publicPath` as well

```
  if (outputPath) {
          // strip the webpack config's output.path
          config.stripPrefix = outputPath + '/';
        }

        if (compiler.options.output.publicPath) {
          // prepend the public path to the resources
          config.replacePrefix = compiler.options.output.publicPath;
        }
```

**Comment by sqal** [comment]
thanks @terjebra This works for me:

```js

new SWPrecachePlugin({
  cacheId: 'MyApp',
  filename: 'service-worker.js',
  staticFileGlobsIgnorePatterns: [/index\.html$/, /\.map$/],
  stripPrefix: /^win/.test(process.platform) ? `${paths.appBuild}/`.replace(/\\/g, '/') : false,
})
`

although webpack still logs wrong path during build

```
Caching static resource "C:/dev/project/build/static/media/logo_167315a9.svg" (1.08 kB)
Caching static resource "C:/dev/project/build/static/js/0.62d33e9c.chunk.js" (36.94 kB)
Caching static resource "C:/dev/project/build/static/js/1.f53ed34c.chunk.js" (72.52 kB)
Caching static resource "C:/dev/project/build/static/js/2.0836cb9f.chunk.js" (50.89 kB)
Caching static resource "C:/dev/project/build/static/js/3.c29ec025.chunk.js" (9.7 kB)
Caching static resource "C:/dev/project/build/static/js/4.a873a95b.chunk.js" (13.39 kB)
Caching static resource "C:/dev/project/build/static/js/5.b6ea2f9a.chunk.js" (6.35 kB)
Caching static resource "C:/dev/project/build/static/js/6.73094e6e.chunk.js" (546 B)
Caching static resource "C:/dev/project/build/static/js/app.acad562e.js" (81.45 kB)
Caching static resource "C:/dev/project/build/static/js/vendor.e0a43902.js" (103.07 kB)```

**Comment by primozs** [comment]
The same bug is on osx.

**Comment by jeffposnick** [comment]
As a heads-up, this was raised in https://github.com/facebookincubator/create-react-app/issues/2235, and seems like it could be a blocking issue for all Windows developers.

If there is an easy fix for the plugin, note that we're using `0.9.1` due to #74, so ideally the fix could be backported as `0.9.2`.

**Comment by goldhand** [comment]
Going to try adding the `stripPrefix: path.join(__dirname, 'dist').replace(/\\/g,"/")` solution to the default configuration. I'm just not aware of all the environments windows users are developing in but I'll give it a go! If there's any window's users who can help on this, would be appreciated :smile:

**Comment by sqal** [comment]
@goldhand that is exact workaround I use in my project:

```
const isWin = /^win/.test(process.platform);

swPrecache.write(`${buildPath}/service-worker.js`, {
  ...
  stripPrefix: isWin ? buildPath.replace(/\\/g, '/') : buildPath,
}, done);
```

tested on windows/mac os :)

**Comment by goldhand** [comment]
Thanks @sqal that looks great 😄

**Comment by goldhand** [comment]
Didn't mean to auto close this. The fix will be in `v0.11.1` but I will also publish a fix in `v0.9.2` for cra

**Comment by goldhand** [comment]
fixed and published in `v0.11.1`
